Dev 6515/investigate paragraph editor tool not formatting content onpaste

benlammers opened the pull request:
- fix parsing for links removing all HTML tags
- remove timeout for loading spinner

Before


https://github.com/discolearn/editor.js/assets/41494387/58055286-b760-4470-97c5-feb8c519d743


After


https://github.com/discolearn/editor.js/assets/41494387/a2e345ea-8d5b-491f-8121-730cde71b873

cam-perry commented on src/components/core.ts:
```diff
@@ -80,27 +80,24 @@ export default class Core {
-           */
-          onReady();
-        // eslint-disable-next-line @typescript-eslint/no-magic-numbers
-        }, 500);
```
🙌

cam-perry commented on src/components/modules/paste.ts:
```diff
@@ -195,10 +195,17 @@ export default class Paste extends Module {
+        return ((word.startsWith('http:')) || (word.startsWith('https:')));
+      });
+
+    for (const link of links) {
+      htmlData = htmlData.replace(link, `<a href="${link}">${link}</a>`);
+    }
```
Issue here: if the same link exists twice in the pasted text you will be replacing the `href` text again like `<a href="<a href="link">link</a>">link</a>`.

It also seems to get messed up by HTML anchor tags that are pasted in with existing links (i.e. paste the same hyperlink from google docs twice).

https://github.com/discolearn/editor.js/assets/33005811/3db995ef-54fb-4eaf-ab22-2d1b2425e639

benlammers replied:
Ah good catch, made it so we ignore any links that are prefixed by `href="` or followed by `</a>`

https://github.com/discolearn/editor.js/assets/41494387/d53ea17f-c3e9-4dc3-af21-ff090229168c

benlammers committed:
Fix parsing for links removing all HTML tags

- instead of replacing HTML data with plain text, get the links from the plain text and replace them in the HTML

benlammers committed:
Remove timeout and loading spinner

- remove timeout that is used to focus a block when preparing the editor

benlammers committed:
Bump version in package.json

cam-perry commented on src/components/modules/paste.ts:
```diff
@@ -195,10 +195,27 @@ export default class Paste extends Module {
+      const escapedLink = link.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+
+      // Match the link if it is not already wrapped in an <a> tag and if it is not an href attribute
+      const regexPattern = new RegExp(`(?<!href=")${escapedLink}(?<!!\\/a>)`, 'g');
```
Regex magic 🧙
